python/tests/scripts/e2_states.metta

```diff
@@ -36,33 +47,49 @@
   (get-type &state-token)
   (StateMonad Expression))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; State atoms have a defined type based on initial state
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; &state-token was initialized with state of type (A B), so
+; it cannot be changed to int
 !(assertEqual
   (change-state! &state-token 1)
   (Error 1 BadType))
+
+; the new state here is int, so it cannot be changed to string
 !(assertEqual
   (change-state! (new-state 1) "S")
   (Error "S" BadType))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ; Changing the content of the state atom
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; (nop ...) provides a way to obtain side-effects of a statement
+; while ignoring unwanted return values.
 !(nop (change-state! &state-token (C D)))
 
 ; The same state atom has different content now
 !(assertEqual
    (get-state (get-token))
    (C D))
 
-
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; MeTTa provides proper encapsulated state manipulation
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; scoping doesn't conflict between let bound variables and state atom types.
 !(assertEqual
   (let $x (new-state 1)
      (change-state! $x (+ (get-state $x) 1)))
   (new-state 2))
 
-; Chcking that there is no conflict between variable names in let-expression
-; and type definition of new-state
+; Checking that state atoms don't conflict with surrounding scopes;
```

Comment:
The problem is `new-state` has (or had?) internal type definition using `$tnso` variable name. New explanation doesn't say this, so it is not clear what it really means.

Reply:
Okay, restoring your original description.